# Carga de librerias

In [10]:
from pymongo import MongoClient
import time
from datetime import datetime, timedelta

client = MongoClient('localhost', 27017)
db = client["test_mongo"]

# Eliminación de documentos de la base de datos con 1 millón de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 1 millón de documentos indexados.

In [2]:
coll = db["extracto_cuenta_1m2"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [4]:
def eliminar_saldo_min(saldo_min):
    return coll.delete_many(
        {
            "saldo":
                {
                    "$lte": saldo_min
                }
        }
)

In [14]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 7 documentos
En un tiempo de 0:00:03.939482


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 3.9394 s       |

-------------------

# Eliminación de documentos de la base de datos con 10 millones de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 10 millones de documentos indexados.

In [16]:
coll = db["extracto_cuenta_10m2"]

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [17]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 111 documentos
En un tiempo de 0:00:33.027268


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 33.0273 s      |

------------------

# Eliminación de documentos de la base de datos con 30 millones de registros y todos los campos indexados

En esta sección se revisará la operación de eliminación en la colección de 1 millón de documentos ya indexados.

In [19]:
coll = db["extracto_cuenta_30m"]


## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [20]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 313 documentos
En un tiempo de 0:01:40.041568


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 1 min 40.0416 s      |

---------------------

# Eliminación de documentos de la base de datos con 10 millones de registros y 3 campos indexados (fecha, descripcion, nota)

En esta sección se revisará la operación de eliminación en la colección de 10 millones de documentos indexados en tres campos: fecha, descripción y nota.

In [5]:
coll = db["extracto_cuenta_10m_3idx"]
coll.index_information()

{'_id_': {'v': 2, 'key': [('_id', 1)]},
 'fecha_1': {'v': 2, 'key': [('fecha', 1)]},
 'descripcion_1': {'v': 2, 'key': [('descripcion', 1)]},
 'nota_text': {'v': 2,
  'key': [('_fts', 'text'), ('_ftsx', 1)],
  'weights': SON([('nota', 1)]),
  'default_language': 'english',
  'language_override': 'language',
  'textIndexVersion': 3}}

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [6]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 111 documentos
En un tiempo de 0:04:07.969962


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 4 min 7.9699   |

-------------------------

# Eliminación de documentos de la base de datos con 30 millones de registros y 3 campos indexados (fecha, descripcion, nota)

En esta sección se revisará la operación de eliminación en la colección de 30 millones de documentos indexados en tres campos: fecha, descripción y nota.

In [8]:
coll = db["extracto_cuenta_30m_3idx"]
coll.index_information()

{'_id_': {'v': 2, 'key': [('_id', 1)]},
 'fecha_1': {'v': 2, 'key': [('fecha', 1)]},
 'descripcion_1': {'v': 2, 'key': [('descripcion', 1)]},
 'nota_text': {'v': 2,
  'key': [('_fts', 'text'), ('_ftsx', 1)],
  'weights': SON([('nota', 1)]),
  'default_language': 'english',
  'language_override': 'language',
  'textIndexVersion': 3}}

## Eliminación 1: Saldo menor

Se eliminarán de la base de datos aquellos documentos cuyos saldos sean menores a determinado número.

In [9]:
saldo_min= 100

start_time = time.time()
results = eliminar_saldo_min(saldo_min)
end_time = time.time()

print(f"Se han eliminado {results.deleted_count} documentos")
print(f"En un tiempo de {str(timedelta(seconds = (end_time - start_time)))}")

Se han eliminado 313 documentos
En un tiempo de 0:17:22.442243


| Tabla resumen de resultados | Valor obtenido |
|-----------------------------|----------------|
| Tiempo eliminación 1        | 17 min 22.4422 s |